homeassistant/components/neato/sensor.py

```diff
@@ -0,0 +1,102 @@
+"""Support for Neato sensors."""
+import logging
+
+from datetime import timedelta
+from pybotvac.exceptions import NeatoRobotException
+
+from homeassistant.components.sensor import DEVICE_CLASS_BATTERY
+from homeassistant.helpers.entity import Entity
+
+from .const import NEATO_ROBOTS, NEATO_LOGIN, NEATO_DOMAIN, SCAN_INTERVAL_MINUTES
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=SCAN_INTERVAL_MINUTES)
+
+BATTERY = "Battery"
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Neato sensor."""
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Neato sensor using config entry."""
+    dev = []
+    for robot in hass.data[NEATO_ROBOTS]:
+        dev.append(NeatoSensor(hass, robot))
+
+    if not dev:
+        return
+
+    _LOGGER.debug("Adding robots for sensors %s", dev)
+    async_add_entities(dev, True)
+
+
+class NeatoSensor(Entity):
+    """Neato sensor."""
+
+    def __init__(self, hass, robot):
+        """Initialize Neato sensor."""
+        self.robot = robot
+        self.neato = hass.data.get(NEATO_LOGIN)
+        self._available = self.neato.logged_in if self.neato is not None else False
+        self._robot_name = f"{self.robot.name} {BATTERY}"
+        self._robot_serial = self.robot.serial
+        self._state = None
+
+    def update(self):
+        """Update Neato Sensor."""
+        if self.neato is None:
+            _LOGGER.error("Error while updating sensor")
+            self._state = None
+            self._available = False
+            return
+
+        try:
+            self.neato.update_robots()
+            self._state = self.robot.state
```

Comment:
Please limit the try block to the code that can raise the exception. The other lines should be moved below after the `return`.

Reply:
@MartinHjelmare `self.robot.state` does IO so this call has potential to fail.  I can move out the variable for setting availability below `return`

Reply:
Good.

Reply:
We can move the line `availability`, but personally I like that there. This line belongs to the two lines above, because these three lines also appear together in the exception block. It doesn't hurt to leave it there and it improves readability. Only my opinion.

Reply:
Ok. We will take a look at the other platforms as well and clean them up according to your reviews

Reply:
ok the variable has been moved out, we will update the rest in a future PR :)